helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -788,39 +791,43 @@ public void setDefaultPartitionWeightMap(Map<String, Integer> weightDataMap)
   private void setDefaultCapacityMap(ClusterConfigProperty capacityPropertyType,
       Map<String, Integer> capacityDataMap) throws IllegalArgumentException {
     if (capacityDataMap == null) {
-      throw new IllegalArgumentException("Default capacity data is null");
+      _record.getMapFields().remove(capacityPropertyType.name());
+    } else {
+      Map<String, String> data = new HashMap<>();
+      capacityDataMap.entrySet().stream().forEach(entry -> {
+        if (entry.getValue() < 0) {
```

Comment:
If the value string is not valid integer representation, will this throw other Exception? If so, do you want to catch it and convert to IllegalArgumentException?

Reply:
The value is not a string, it is Integer.